### PR TITLE
ILP64 reference library for LAPACK on Windows (#917)

### PR DESCRIPTION
**Pull request by daineAMD**
hotfix into 6.3.

Local windows tests passed with rocSOLVER.

### CODE REVIEW

**Review by TorreZuk — APPROVED** [review]
Looks good to me, but probably need to get solver team member to review.
Did they add ILP64?   
do you want a copilot to review too, maybe we should start asking for one file in each PR?

**Comment by daineAMD** [comment]
I'll plan on merging this by the end of the day today. @qjojo @cgmb this PR relates to reference code for solver functions so may be of interest for you to review.

**Comment by daineAMD** [comment]
Note sure why there are a couple trmm failures on the CUDA backend. They are unrelated to this PR as [they were seen on develop as well](http://math-ci.amd.com/blue/organizations/jenkins/main%2Fprecheckin-cuda%2FhipBLAS/detail/develop/346/pipeline/249). I'll try again to reproduce them today.

File: clients/include/solver/testing_getrf.hpp [hunk 1]
```diff
@@ -95,12 +95,12 @@ void testing_getrf(const Arguments& arg)
...
     // Naming: dK is in GPU (device) memory. hK is in CPU (host) memory
-    host_matrix<T>   hA(M, N, lda);
-    host_matrix<T>   hA1(M, N, lda);
-    host_matrix<int> hIpiv(1, Ipiv_size, 1);
-    host_matrix<int> hIpiv1(1, Ipiv_size, 1);
-    host_vector<int> hInfo(1);
-    host_vector<int> hInfo1(1);
+    host_matrix<T>       hA(M, N, lda);
+    host_matrix<T>       hA1(M, N, lda);
+    host_matrix<int>     hIpiv(1, Ipiv_size, 1);
```
Comment (NaveenElumalaiAMD) [inline clients/include/solver/testing_getrf.hpp:100]: int64_t here? or Is it only for LAPACK ?

File: clients/include/solver/testing_getri_batched.hpp [hunk 2]
```diff
@@ -205,10 +205,13 @@ void testing_getri_batched(const Arguments& arg)
         }
 
         // perform LU factorization on A
-        int* hIpivb = hIpiv.data() + b * strideP;
-        hInfo[b]    = ref_getrf(M, N, hA[b], lda, hIpivb);
+        int64_t* hIpivb = hIpiv64.data() + b * strideP;
+        hInfo[b]        = ref_getrf(M, N, hA[b], lda, hIpivb);
     }
 
+    for(int i = 0; i < Ipiv_size; i++)
```
Comment (NaveenElumalaiAMD) [inline clients/include/solver/testing_getri_batched.hpp:212]: int64_t?

File: clients/include/solver/testing_getrs_strided_batched.hpp [hunk 3]
```diff
@@ -89,10 +91,13 @@ void setup_getrs_strided_batched_testing(const Arguments&                arg,
...
     }
 
+    for(int i = 0; i < Ipiv_size; i++)
+        hIpiv32[i] = hIpiv64[i];
+
     // Copy data from CPU to device
     CHECK_HIP_ERROR(dA.transfer_from(hA));
     CHECK_HIP_ERROR(dB.transfer_from(hB));
-    CHECK_HIP_ERROR(hipMemcpy(dIpiv, hIpiv, Ipiv_size * sizeof(int), hipMemcpyHostToDevice));
+    CHECK_HIP_ERROR(hipMemcpy(dIpiv, hIpiv32, Ipiv_size * sizeof(int), hipMemcpyHostToDevice));
```
Comment (NaveenElumalaiAMD) [inline clients/include/solver/testing_getrs_strided_batched.hpp:100]: I know here it is `hIpiv32` is that same convention going to followed in all the other functions as well? To differentiate between `hIpiv32` and `hIpiv64`

**Review by NaveenElumalaiAMD — APPROVED** [review]

File: clients/include/solver/testing_getri_batched.hpp (hunk 2 above)
Comment (daineAMD) [reply clients/include/solver/testing_getri_batched.hpp:212]: Yeah. There will have to be a fair amount of changes to these files once rocSOLVER 64-bit backend is supported. We previously changed to use 64-bit reference code, so these changes are just to ensure we're passing in 64-bit params to the reference code. So I think this is fine for now as it'll be addressed once we actually support 64-bit testing.

File: clients/include/solver/testing_getrs_strided_batched.hpp (hunk 3 above)
Comment (daineAMD) [reply clients/include/solver/testing_getrs_strided_batched.hpp:100]: Long-term we could probably just use 64-bit here and wouldn't need the duplicates I think. Just made as minimal changes here and will add the full support at a later point.

File: clients/include/solver/testing_getrf.hpp (hunk 1 above)
Comment (daineAMD) [reply clients/include/solver/testing_getrf.hpp:100]: Yeah the 32-bit is needed right now for rocSOLVER interface, 64-bit for reference-code interface

**Pull request merged**